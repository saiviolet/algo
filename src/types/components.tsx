import {ElementStates} from "./element-states";

export interface ILetter {
  letter: string;
  key: string;
  state: ElementStates;
}

export interface IStateString {
  buttonLoader: boolean;
  buttonDisabled: boolean;
  inputValue: string;
  string: ILetter[] | null;
}

export interface IStateFibonacci {
  buttonLoader: boolean;
  buttonDisabled: boolean;
  inputValue: string;
  number: number | null;
  array: number[];
}

export type TFibonacci = (n: number, updateState: React.Dispatch<any>) => void;
