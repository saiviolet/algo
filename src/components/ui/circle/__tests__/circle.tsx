import {Circle} from "../circle";
import { render } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import {ElementStates} from "../../../../types/element-states";

const testingData = {
  letter: 'test',
  headString: 'head',
  headJSX: <Circle letter={'test head'}/>,
  tailString: 'tail',
  tailJSX: <Circle letter={'test tail'}/>,
  index: 22,
  isSmall: true,
  defaultState: ElementStates.Default,
  modifiedState: ElementStates.Modified,
  changingState: ElementStates.Changing,
}

describe('( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle',() => {

  test('Отрисовка  пустого компонента (без буквы)', () => {
    const {asFragment} = render(<Circle />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Отрисовка компонента с буквой', () => {
    const testRenderer = TestRenderer.create(<Circle letter={testingData.letter}/>);
    const test = testRenderer.root;
    expect(test.props.letter.length).toBeLessThan(5);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  test('Отрисовка компонента с текстовым head', () => {
    const circle = render(<Circle head={testingData.headString}/>);
    expect(circle.getByText('head')).toBeInTheDocument();
    expect(circle.asFragment()).toMatchSnapshot();
  });
  test('Отрисовка компонента с react-элементом в head;', () => {
    const circle = render(<Circle head={testingData.headJSX}/>);
    expect(circle.getByText(/test head/i)).toBeInTheDocument();
    expect(circle.asFragment()).toMatchSnapshot();
  });
  test('Отрисовка компонента с текстовым tail', () => {
    const circle = render(<Circle tail={testingData.tailString}/>);
    expect(circle.getByText('tail')).toBeInTheDocument();
    expect(circle.asFragment()).toMatchSnapshot();
  });
  test('Отрисовка компонента с react-элементом в tail;', () => {
    const circle = render(<Circle tail={testingData.tailJSX}/>);
    expect(circle.getByText(/test tail/i)).toBeInTheDocument();
    expect(circle.asFragment()).toMatchSnapshot();
  });
  test('Отрисовка компонента c index', () => {
    const circle = render(<Circle index={testingData.index}/>);
    expect(circle.getByText(/22/i)).toBeInTheDocument();
    expect(circle.asFragment()).toMatchSnapshot();
  });
  test('Отрисовка компонента с пропсом isSmall', () => {
    const testRenderer = TestRenderer.create(<Circle isSmall={testingData.isSmall}/>);
    const test = testRenderer.root;
    expect(test.props.isSmall).toBeTruthy();
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  test('Отрисовка компонента с default state', () => {
    const testRenderer = TestRenderer.create(<Circle state={testingData.defaultState}/>);
    const test = testRenderer.root;
    expect(test.props.state).toBe('default');
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  test('Отрисовка компонента с modified state', () => {
    const testRenderer = TestRenderer.create(<Circle state={testingData.modifiedState}/>);
    const test = testRenderer.root;
    expect(test.props.state).toBe('modified');
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  test('Отрисовка компонента с changing state', () => {
    const testRenderer = TestRenderer.create(<Circle state={testingData.changingState}/>);
    const test = testRenderer.root;
    expect(test.props.state).toBe('changing');
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

})