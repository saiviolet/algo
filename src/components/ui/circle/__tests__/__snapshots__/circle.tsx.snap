// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка  пустого компонента (без буквы) 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-test="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента c index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-test="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      22
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</DocumentFragment>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-test="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-test="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с react-элементом в head; 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
          data-test="circle"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            test head
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
      data-test="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с react-элементом в tail; 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-test="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
          data-test="circle"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            test tail
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с буквой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с пропсом isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-test="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с текстовым head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      head
    </div>
    <div
      class="circle   default"
      data-test="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`( ͡⊙ ͜ʖ ͡⊙)Тесты компонента Circle Отрисовка компонента с текстовым tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
      data-test="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      tail
    </div>
  </div>
</DocumentFragment>
`;
